src/Data/List/Compat.hs

```diff
@@ -1,9 +1,11 @@
 module Data.List.Compat (
   module Base
```

Comment:
Only re-exporting `Data.List` for the latest version of `base` gives us the guarantee that the `Data.List.Compat` module is indeed exactly the same as `Data.List`.  This is a nice thing, as we can use `Data.List.Compat` with the latest version of base as a reference for testing.